src/forwarddiff.jl

```diff
@@ -0,0 +1,51 @@
+using .ForwardDiff: ForwardDiff, DiffResults, StaticArrays
+
+struct ForwardDiffBackend <: AbstractBackend end
+
+@primitive function pushforward_function(ba::ForwardDiffBackend, f, xs...)
```

Comment:
Why do we need both this and the next one? Can we just use `x` here instead of `xs...`?

Reply:
can check if it's a tuple and unwrap it